src/monitormanager.cpp

```diff
@@ -151,6 +154,63 @@ void MonitorManager::relayoutTag(HSTag *tag)
     }
 }
 
+int MonitorManager::removeMonitor(Input input, Output output)
+{
+    if (input.size() < 2) {
+        return HERBST_NEED_MORE_ARGS;
+    }
+
+    input.shift();
+    string monitorIdxString = input.front();
+    auto monitor = byString(monitorIdxString);
+
+    if (monitor == nullptr) {
+        output << input.command() << ": Monitor \"" << monitorIdxString << "\" not found!\n";
+        return HERBST_INVALID_ARGUMENT;
+    }
+
+    if (size() <= 1) {
+        output << input.command() << ": Can't remove the last monitor\n";
+        return HERBST_FORBIDDEN;
+    }
+
+    removeMonitor(monitor);
+
+    return HERBST_EXIT_SUCCESS;
+}
+
+void MonitorManager::removeMonitor(HSMonitor* monitor)
+{
+    auto monitorIdx = index_of(monitor);
+
+    if (g_cur_monitor > index_of(monitor)) {
+        // Take into account that the current monitor will have a new
+        // index after removal:
+        g_cur_monitor--;
+    }
+
+    // Hide all clients visible in monitor
+    assert(monitor->tag != nullptr);
+    assert(monitor->tag->frame != nullptr);
+    monitor->tag->frame->setVisibleRecursive(false);
+
+    g_monitors->removeIndexed(monitorIdx);
```

Comment:
Around here and as well as in the Monitor destructor I changed things a bit and I'm not entirely sure if this is correct. Please double-check.

Reply:
As far as I can see, the order is fine, or even better. (Before, `stack_remove_slice` was called twice...)